lib/signed-xml.js

```diff
@@ -333,28 +333,56 @@ SignedXml.defaultNsForPrefix = {
 
 SignedXml.findAncestorNs = findAncestorNs;
 
-SignedXml.prototype.checkSignature = function(xml) {
+SignedXml.prototype.checkSignature = function(xml, callback) {
+  if (callback != null && typeof callback !== 'function') {
+    throw new Error("Last paramater must be a callback function")
+  }
+
   this.validationErrors = []
   this.signedXml = xml
 
   if (!this.keyInfoProvider) {
-    throw new Error("cannot validate signature since no key info resolver was provided")
+    var err = new Error("cannot validate signature since no key info resolver was provided")
+    if (!callback){
+      throw err
+    }else{
+      callback(err)
+      return
+    }
   }
 
   this.signingKey = this.keyInfoProvider.getKey(this.keyInfo)
-  if (!this.signingKey) throw new Error("key info provider could not resolve key info " + this.keyInfo)
+  if (!this.signingKey) {
+    var err = new Error("key info provider could not resolve key info " + this.keyInfo)
+    if (!callback){
+      throw err
+    }else{
+      callback(err)
+      return
+    }
+  }
 
   var doc = new Dom().parseFromString(xml)
 
   if (!this.validateReferences(doc)) {
-    return false;
-  }
-  
-  if (!this.validateSignatureValue(doc)) {
-    return false;
+    if (!callback){
+      return false;
+    }else{
+      callback(new Error('Could not validate references'))
+    }
   }
 
-  return true
+  var ret = this.validateSignatureValue(doc)
+  if (ret instanceof Promise){
```

Comment:
I wish to avoid Promises because I'm aware that xml-crypto is used in environments where getting the latest versions of node is a significant bureaucratic burden. To that end, I'm trying my best to support even versions of Node that are no longer maintained